app/controllers/v3/service_instances_controller.rb

```diff
@@ -431,7 +428,15 @@ def unprocessable_space!
     unprocessable!('Invalid space. Ensure that the space exists and you have access to it.')
   end
 
-  def unprocessable_service_plan!
+  def service_plan_does_not_exist!
+    logger.info('Service Plan does not exist.')
+    unprocessable!('Invalid service plan. Ensure that the service plan exists, is available, and you have access to it.')
+  end
+
+  def service_plan_not_visible_to_user!(service_plan)
+    user = VCAP::CloudController::SecurityContext.current_user
+    logger.info("Service Plan with guid '#{service_plan.guid}' and id '#{service_plan.id}' is not visible in current org to user with guid '#{user.guid}'.")
```

Comment:
'in current org' is not correct here as the user can see the plan if he has access to it in any org